opensearch-project/opensearch-build
# Support new distribution-validations and add few enhancements to validation workflow

### Description
1. Support Zip and Deb validations.
2. Move the download_artifacts method from child classes(except docker) to parent class - validation.py.
3. Changes to validation_args.py to fix the bug https://github.com/opensearch-project/opensearch-build/issues/4441
4. Cluster readiness is checked by requesting the api's instead of adding sleep().
5. Changes to docker-validation:
    1. Add security strong password changes before running the container.
    2. Bring up the services opensearchnode1 and opensearchnode2 only when OS is validated, bringup all the services in 
        compose file when both OS and OSD are provided.
    3. Pull and validate the latest staging docker image in the given version if build number is not provided.
    
Addresses #4424 and #4423

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
